src/nifreeze/analysis/measure_agreement.py

```diff
@@ -0,0 +1,226 @@
+    """Compute quantities of interest for the Bland-Altman plot.
+
+    Parameters
+    ----------
+    data1 : :obj:`numpy.ndarray`
+        Data values.
+    data2 : :obj:`numpy.ndarray`
+        Data values.
+    ci : :obj:`float`
+        Confidence interval size. Must be in the [0, 1] range.
+
+    Returns
+    -------
+    diff : :obj:`numpy.ndarray`
+        Differences.
+    mean_diff : :obj:`numpy.ndarray`
+        Mean differences.
+    mean : :obj:`numpy.ndarray`
+        Mean values (across both data arrays).
+    std_diff : :obj:`numpy.ndarray`
+        Standard deviation of differences.
+    loa_lower : :obj:`numpy.ndarray`
+        Lower limit of agreement.
+    loa_upper : :obj:`numpy.ndarray`
+        Upper limit of agreement.
+    ci_mean : :obj:`numpy.ndarray`
+        Confidence interval of mean values.
+    ci_loa : :obj:`numpy.ndarray`
+        Confidence interval of limits of agreement.
+    """
+
+    _check_bland_altman_data(data1, data2)
+    _check_ci(ci)
+
+    # Compute mean
+    mean = np.mean([data1, data2], axis=0)
+
+    # Compute differences, mean difference, and std dev of differences
+    diff = data1 - data2
+    mean_diff = np.mean(diff)
+    std_diff = np.std(diff, axis=0)
+    # std_diff = np.std(diff, ddof=1)  # Use Bessel's correction
+
+    # Compute confidence interval limits of agreement (LoA)
+    z_score = compute_z_score(ci)
+    loa_lower = mean_diff - z_score * std_diff
+    loa_upper = mean_diff + z_score * std_diff
+
+    n = len(diff)
+
+    # Compute t-value for the confidence intervals (CIs)
+    t_val = stats.t.ppf((1 + ci) / 2, n - 1)
+
+    # Compute confidence intervals
+
+    # Confidence interval for the mean difference
+    std_err_mean = std_diff / np.sqrt(n)
+    ci_mean = t_val * std_err_mean
+
+    # Confidence interval for the LoA
+    # Follows Bland-Altman 1999 and Altman 1983, where the standard error of LoA
+    # SE_{LoA} = \sigma_{d} \ sqrt{2/n}
+    # The confidence interval (CI) for LoA is then calculated using the
+    # t-distribution critical value.
+    std_err_loa = std_diff * np.sqrt(2 / n)
+    ci_loa = t_val * std_err_loa
+
+    return diff, mean_diff, mean, std_diff, loa_lower, loa_upper, ci_mean, ci_loa
+
+
+def get_reliabilty_mask(diff, loa_lower, loa_upper):
+
+    # Filter data within the CI
+    return (diff >= loa_lower) & (diff <= loa_upper)
+
+
+def identify_bland_altman_saliency(data1, data2, ci, top_n=100):
+
+    (
+        diff,
+        mean_diff,
+        mean,
+        std_diff,
+        loa_lower,
+        loa_upper,
+        _,
+        _,
+    ) = compute_bland_altman_features(data1, data2, ci)
+
+    # Filter data within the CI
+    reliable_mask = get_reliabilty_mask(diff, loa_lower, loa_upper)
+    reliability_idx = np.where(reliable_mask)[0]
+    reliable_mean = mean[reliable_mask]
+
+    # Compute the ratio between max and median of mean values
+    median_mean = np.median(reliable_mean)
+    mean_ratios = reliable_mean / median_mean
```

Comment:
This needs to be reviewed. I do not think my understanding of the points that we seek here is correct.